src/compiler/tracing.ts

```diff
@@ -1,16 +1,24 @@
 /*@internal*/
 /** Tracing events for the compiler. */
 namespace ts.tracing {
+    export const enum Mode {
+        Project,
+        Build,
+        Server,
+    }
+
     let fs: typeof import("fs") | false | undefined;
 
     let traceCount = 0;
     let traceFd: number | undefined;
 
+    let mode: Mode = Mode.Project; // This initial value should never be consumed

```

Comment:
I don't understand the purpose of `Project` given that you're saying that it shouldn't be consumed...  (Since there is code that calls it with that value.)

But if this should be initially in some state that is invalid, then maybe add some `None`, or `Unset` or similar?

Reply:
`Project` is a valid value - the comment was intended to convey that the variable is effectively uninitialized.

Reply:
Just drop the initializer.